parsl/app/bash.py

```diff
@@ -160,3 +168,203 @@ def __call__(self, *args, **kwargs):
                              app_kwargs=invocation_kwargs)
 
         return app_fut
+
+
+class BashWatcher(FileSystemEventHandler):
+    """A class to watch for file creation events on local file systems"""
+
+    def __init__(self):
+        super().__init__()
+        self.added_files = []
+
+    def on_created(self, event) -> None:
+        if event.is_directory:
+            return
+        self.added_files.append(event.src_path)
+
+    def on_deleted(self, event) -> None:
+        """ If a file is deleted, and we recoded its creation, remove it from our list as well"""
+        if event.is_directory:
+            return
+        if event.src_path in self.added_files:
+            self.added_files.remove(event.src_path)
+
+    def on_moved(self, event) -> None:
+        """ If a file is moved, and we recorded its creation, change its list entry"""
+        if event.is_directory:
+            return
+        if event.src_path in self.added_files:
+            self.added_files.remove(event.src_path)
+            self.added_files.append(event.dest_path)
+
+
+class BashTracker:
+    """A class to watch for file creations on networked file systems. This is done by snapshotting the monitored
+    directories before and after and recording the difference"""
+
+    def __init__(self):
+        self.paths = []
+        self.files = []
+        self.added_files = []
+
+    def add_path(self, path: str) -> None:
+        """Add a directory to be monitored
+
+        Parameters
+        ----------
+        path: str
+            The path to be monitored
+        """
+        self.paths.append(path)
+        self.files.extend(glob.glob(os.path.join(path, '**', '*'), recursive=True))
+
+    def update(self) -> None:
+        """Generate the list of new files based on the initial snapshot and the current state of the paths to be
+         monitored."""
+        temp = []
+        for pth in self.paths:
+            temp.extend(glob.glob(os.path.join(pth, '**', '*'), recursive=True))
+        updated = set(temp)
+        self.added_files = list(updated - set(self.files))
+
+
+class BashObserver(Observer):
```

Comment:
@astro-friedel Based on our discussion at the parsl dev calls, I was under the impression that the plan was to simplify the mechanism for detecting app specific files. We'd discussed capturing the delta of the files in a directory path before and after executing the command, as being a much lighter solution than relying on continuous tracking. Let me know what you think.